control-plane/typeSpecConverter.cpp

```diff
@@ -111,6 +115,49 @@ bool TypeSpecConverter::preorder(const IR::Type_Name* type) {
     return false;
 }
 
+bool TypeSpecConverter::preorder(const IR::Type_Newtype* type) {
+    if (p4RtTypeInfo) {
+        bool orig = true;
+        cstring uri;
+        uint32_t sdnB;
+        auto ann = type->getAnnotation(IR::Annotation::p4runtimeTranslationAnnotation);
+        if (ann != nullptr) {
+            orig = false;
+            int i = 0;
+            for (auto a : ann->body) {
+                if (!i++)
+                    uri = a->text;
+                if (isdigit(a->text.c_str()[0])) {
+                    sdnB = atoi(a->text);
+                }
+            }
+        }
+
+        auto name = std::string(type->controlPlaneName());
+        auto types = p4RtTypeInfo->mutable_new_types();
+        if (types->find(name) == types->end()) {
+            auto newTypeSpec = new p4configv1::P4NewTypeSpec();
+            auto newType = type->type;
+            auto n = newType->to<IR::Type_Name>();
+            visit(n);
+            auto typeSpec = map.at(n);
+            if (orig) {
+                (void)newTypeSpec->mutable_original_type();
+                auto dataType = newTypeSpec->mutable_original_type();
+                dataType->mutable_bitstring()->CopyFrom(typeSpec->bitstring());
```

Comment:
how do you know this is a bitstring? You should at least add a check. Personally I think that `newTypeSpec->mutable_original_type()->CopyFrom(typeSpec)` is both simpler and more correct.

Reply:
I add a check for bitstring first.  Your suggested code does not work - didn't get time to debug why.  It is used with `(*typeSpec)`.  Later, when I get time, I may debug this one.